src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,156 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+
+        if installation_substatus_index is not None:    # If installation data exists
+            self.__installation_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, installation_substatus_index)
+            self.__installation_patches_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_patches_copy)
+
+        status_file_without_patches_size_in_bytes = self.__size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # Deepcopy, fully copies the object to avoid reference modifications
+
+        max_allowed_patches_size_in_bytes = Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES - status_file_without_patches_size_in_bytes
+        self.composite_logger.log_debug("Status file limits evaluated. [FileSizeWithoutPatchesInBytes={0}] [MaxAllowedPatchesSizeInBytes={1}]".format(
+            str(status_file_without_patches_size_in_bytes), str(max_allowed_patches_size_in_bytes)))
+
+        while status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:
+            # Start truncation process
+            patches_retained_in_assessment, self.__assessment_patches_removed, patches_retained_in_installation, self.__installation_patches_removed = \
+                self.__start_truncation_process(self.__assessment_patches_copy, self.__installation_patches_copy, max_allowed_patches_size_in_bytes, low_pri_index)
+
+            if len(self.__assessment_patches_removed) > 0:
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_ASSESSMENT_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_patches=patches_retained_in_assessment,
+                    substatus_message=self.__assessment_substatus_msg_copy, substatus_index=assessment_substatus_index)
+
+            if len(self.__installation_patches_removed) > 0:
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_INSTALLATION_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_patches=patches_retained_in_installation,
+                    substatus_message=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_size_diff_in_bytes = (status_file_size_in_bytes - Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)  # The byte difference is tombstone, new errors, and escape chars byte size
```

Comment:
If we are reducing status file size to below Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES, by truncation (until line 904):

won't status_file_size_in_bytes < Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES

status_file_size_diff_in_bytes is a negative value at this point then.

You have added this considering tombstone and error details which will come in future. But those changes are not in this PR, thus this value is negative at this point.

This is a reason why exclusivity between PRs is very important, we cannot code with some of the future changes incompletely added in the current one.

Please remove lines 905-906 as they are not in use in this PR

Reply:
we are not reducing the status_file_size_in_bytes, we are checking whether post truncation status_file_size_in_bytes is still greater than 126kb, if it is we need to reduce max_allowed_patches_size_in_bytes because of newly added tombstone, errors etc.. 

the code is removed, done.